client/my-sites/upgrades/cart/cart-item.jsx

```diff
@@ -51,6 +52,27 @@ module.exports = React.createClass( {
 		} );
 	},
 
+	monthlyPrice: function() {
+		const { cost, product_slug: productSlug, currency } = this.props.cartItem,
+			// A hacky way to check if we are in signup, since for some reason cartItem.extra.context
+			// is missing in that case (it seems cartItem.extra is overwritten somewhere between)
+			isInSignup = this.props.cartItem.extra && this.props.cartItem.extra.context !== 'calypstore',
```

Comment:
When a new user completes signup, they're redirected to checkout via a POST to `wp-login.php?redirectTo=[checkout_url]`, and the cart is fetched from the server. My guess here is that `extra.context` isn't included when the user enters checkout and the cart is fetched.

As for determining if we're in the checkout post-signup, this seems like an appropriate place to use a query parameter. You could append `?from=signup` to the checkout route the user is redirected to from signup, and pass that `from` value down from the controller to `CartItem` through props.